app/scripts/controllers/transactions/index.js

```diff
@@ -365,28 +365,36 @@ class TransactionController extends EventEmitter {
       return
     }
     this.inProcessOfSigning.add(txId)
-    let nonceLock
+    let nonceLock = {}
     try {
       // approve
       this.txStateManager.setTxStatusApproved(txId)
       // get next nonce
       const txMeta = this.txStateManager.getTx(txId)
       const fromAddress = txMeta.txParams.from
       // wait for a nonce
-      nonceLock = await this.nonceTracker.getNonceLock(fromAddress)
+      const { customNonceValue = null } = txMeta
+      if (!customNonceValue) {
+        nonceLock = await this.nonceTracker.getNonceLock(fromAddress)
+      }
       // add nonce to txParams
       // if txMeta has lastGasPrice then it is a retry at same nonce with higher
       // gas price transaction and their for the nonce should not be calculated
       const nonce = txMeta.lastGasPrice ? txMeta.txParams.nonce : nonceLock.nextNonce
-      txMeta.txParams.nonce = ethUtil.addHexPrefix(nonce.toString(16))
+      const customOrNonce = customNonceValue || nonce
+      txMeta.txParams.nonce = ethUtil.addHexPrefix(customOrNonce.toString(16))
```

Comment:
is there some where we check that the custom nonce is not a NaN?

Reply:
I believe it's impossible for it to be NaN. It comes from an `<input type="number">` field, and it's set to an empty string if it's not a positive number.